qiskit/providers/aer/backends/qasm_simulator.py

```diff
@@ -19,7 +19,7 @@
 from qiskit.providers.models import QasmBackendConfiguration
 from .aerbackend import AerBackend
 # pylint: disable=import-error
-from .qasm_controller_wrapper import qasm_controller_execute
+from .qasm_controller_wrapper import qasm_controller_execute_new
```

Comment:
no `_new` postfix, we will entirely replace `qasm_controller_execute` because the change should has no side effects, right?

Reply:
yep, very true - I plan to adjust this in some way but I was wondering how the dependency of pybind should be added 
It could be done such that both cython and pybind wrappers are generated as separate modules - this could be useful for testing, comparing the two implementations, as well as introducing the new dependency in a 'soft' way - i.e. not breaking the builds of everybody who is currently building with cython and does not have pybind installed

Reply:
I guess that if all our test suite pass with the new pybind backends, everything should be fine. As the interface with Python is the same no matter what cython or pybind version of the backend we are using, let's keep the Cython files for a while and add a flag to the build system so we can build the cython version whenever we want to, thoughts?
```
python ./setup.py bdist_wheel -- -DCYTHON_BACKENDS=True
```
I can implement this and PR your branch.

Reply:
Shoot actually the interfaces aren't the same - *_control_execute_new doesn't take in a string (like the cython ones) right now - although I can probably add a relatively simple 'if isinstance(string)' to the pybind wrappers to sub-set the cython functionality (at hopefully little cost to performance)

Reply:
The pybind dependency is extremely lightweight - pip/conda install pybind11 in probably 90% of cases and a simple fix in any others - so I'm going to just remove the cython bindings - I did add *_controller_execute_json method in case we ever wanted to use the json string interface (and renamed the new one to *_controller_execute)